### PR TITLE
update nats dependencies to modern versions

### PR DESCRIPTION
**Pull request by ripienaar**
I would like to make some follow up PRs to support some of the modern NATS 2.0 security
features and thought I'd start by updating dependencies first before doing those changes

### Required for all PRs:

- [x] Signed [CLA](https://influxdata.com/community/cla/).
- [x] Associated README.md updated.
- [ ] Has appropriate unit tests.

### CODE REVIEW

**Comment by danielnelson** [comment]
Thanks @ripienaar, interested to see the new features, just go ahead and add your changes on top of this changeset and we can merge them in one go.

**Comment by ripienaar** [comment]
@danielnelson OK am about done here, mainly I added `credentials` which is how clients authenticate to new style NATS accounts system.

Additionally moved away from the now deprecated method of connecting via opts.

File: plugins/outputs/nats/README.md [hunk 1]
```diff
@@ -9,6 +9,8 @@ This plugin writes to a (list of) specified NATS instance(s).
   ## Optional credentials
   # username = ""
   # password = ""
+  # credentials = "/etc/telegraf/nats.creds
+
```
Comment (danielnelson) [inline plugins/outputs/nats/README.md]: Can you add this example to the `nats_consumer` README?

**Comment by ripienaar** [comment]
I've updated the comments etc, it's ready to merge if you agree

**Pull request merged**